### ISSUE TITLE
Newsreader: Generate back-link even if the selected overview page is "index"

### ISSUE THREAD

**Issue by PfefferStift** [contao/contao#6316]
### Affected version(s)

Contao 4.13

### Description

**Frontend module "Newsreader":**
When you select the start page (with alias "index") as the **Overview page**, the link "Back to the news overview" is not generated.
I guess, the problem is: _mod_newsreader.html5_ checks if ($this->referer) is empty. And if the index page is selected, the referer is empty.

Contao 4.13, PHP 8.1

**Comment by ausi** [comment]
This should already be fixed in Contao 5, as we use path-absolute URLs there.

I think we need to replace https://github.com/contao/contao/blob/ac7d1a8baf907a5a640e0c68b7111db49eae8915/news-bundle/src/Resources/contao/templates/modules/mod_newsreader.html5#L7 with 
```php
<?php if (null !== $this->referer): ?>
```

**Comment by leofeyer** [comment]
But wouldn‘t that result in an empty `href=""` attribute?

**Comment by fritzmg** [comment]
> But wouldn‘t that result in an empty `href=""` attribute?

Yes, that's fine though due to the `<base href>`.

**Comment by aschempp** [comment]
Or we simply use `<?= $this->referer ?: './' ?>` as we do in other places 😊

**Comment by ausi** [comment]
> Or we simply use `<?= $this->referer ?: './' ?>` as we do in other places 😊

How would that work? The back link would then always be there, event if no overview page was selected.

**Comment by asaage** [comment]
I'm still not completely sold on the overview-Page field in the reader.
see: https://github.com/contao/contao/issues/3586#issuecomment-969283309

```<?= $this->referer ?: './' ?>``` and a boolean/checkbox -field to display a link (or not) might be a decent solution.

**Comment by ausi** [comment]
> I'm still not completely sold on the overview-Page field in the reader.

This is unrelated to this issue. This issue is about a missing back link if you select the home page as the “overview page”.

> `<?= $this->referer ?: './' ?>` and a boolean/checkbox -field to display a link (or not) might be a decent solution.

An extra checkbox is not needed. The “back to overview” link should be shown if a page was selected, otherwise it should not be shown.

**Comment by asaage** [comment]
I'm just saying if you use referer anyway instead of overviewPage there is no need to select a page.
I was thinking of the actual request-header until i saw this: https://github.com/contao/contao/blob/ac7d1a8baf907a5a640e0c68b7111db49eae8915/news-bundle/src/Resources/contao/modules/ModuleNewsReader.php#L88

**Comment by aschempp** [comment]
```php
$this->Template->referer = PageModel::findById($this->overviewPage)->getFrontendUrl() ?: './';
```

should be an easy fix?

**Comment by ausi** [comment]
> ```php
> $this->Template->referer = PageModel::findById($this->overviewPage)->getFrontendUrl() ?: './';
> ```
> should be an easy fix?

Yes I think this is the right fix, as it does not require template modifications.
_(but this should only be fixed in 4.13 not merged into Contao 5)_

**Comment by aschempp** [comment]
> (but this should only be fixed in 4.13 not merged into Contao 5)

why is that?

**Comment by fritzmg** [comment]
In Contao 5 generated URLs are already path absolute and not relative to the base.

**Comment by fritzmg** [comment]
This is automatically fixed in Contao 5 meanwhile.
